shared/lib_shared_inverter.cpp

```diff
@@ -211,66 +210,68 @@ void SharedInverter::calculateACPower(const double powerDC_kW_in, const double D
     Tdry_C = tempC;
     StringV = DCStringVoltage;
     double tempLoss = 0.0;
-	if (m_tempEnabled) {
-		calculateTempDerate(DCStringVoltage, tempC, powerAC_Watts, efficiencyAC, tempLoss);
-	}
-
-	// Convert units to kW- no need to scale to system size because passed in as power to total number of inverters
-	powerDC_kW = powerDC_Watts * util::watt_to_kilowatt;
-	convertOutputsToKWandScale(tempLoss, powerAC_Watts);
-
-	// In event shared inverter is charging a battery only, need to re-convert to negative power
-	if (negativePower) {
-		powerAC_kW = -1.0 * fabs(powerAC_kW);
-	}
+    double power_ratio = 1.0;
+    if (m_tempEnabled) {
+        calculateTempDerate(DCStringVoltage, tempC, powerAC_Watts, power_ratio, tempLoss);
```

Comment:
Since this code is now working to potentially clip the input DC power, this should be done before the calls to `acpower`.